tests/error/inout_errors/subscript_not_setable.err

```diff
@@ -0,0 +1,7 @@
+Guppy compilation failed. Error in file $FILE:24
+
+22:    @guppy(module)
+23:    def test(c: MyImmutableContainer) -> MyImmutableContainer:
+24:        foo(c[0])
+               ^^^^
+GuppyTypeError: Expression of type `MyImmutableContainer` is not assignable since it does not implement the `__setitem__` method
```

Comment:
The error message seems confusing here as the effect of `c[0]` is not an _assignment_. To me it reads more like `__getitem__`. Isn't the error really because `foo` consumes an `inout` and doesn't return it back?

Assuming that is allowed if `__setitem__` is defined for the container, perhaps the error messaging needs to be improved here. Something about `foo` requiring an assignment that's not implemented by the container.

Reply:
The problem is that once the `inout` value has been given back by `foo`, we cannot assign it back to `c[0]` since `c` doesn't implement `__setitem__`.

But you're right, the current message is misleading. I'll update it 👍

Reply:
Ah, that explains the reassignment, certainly better error message will help here.